iceoryx_utils/include/iceoryx_utils/internal/cxx/expected.inl

```diff
@@ -127,96 +153,119 @@ inline bool expected<ValueType, ErrorType>::has_error() const noexcept
 template <typename ValueType, typename ErrorType>
 inline const ErrorType&& expected<ValueType, ErrorType>::get_error() const&& noexcept
 {
-    return std::move(*m_store.template get_at_index<1>());
+    if (!has_undefined_state())
+    {
+        return std::move(*m_store.template get_at_index<1>());
+    }
+    else
+    {
+        std::cerr << "get_error() called on invalid object" << std::endl;
+        std::terminate();
+    }
 }
 
 template <typename ValueType, typename ErrorType>
     inline ErrorType&& expected<ValueType, ErrorType>::get_error() && noexcept
 {
-    return std::move(*m_store.template get_at_index<1>());
+    if (!has_undefined_state())
+    {
+        return std::move(*m_store.template get_at_index<1>());
+    }
+    else
+    {
+        std::cerr << "get_error() called on invalid object" << std::endl;
+        std::terminate();
+    }
 }
 
 template <typename ValueType, typename ErrorType>
     inline ErrorType& expected<ValueType, ErrorType>::get_error() & noexcept
 {
-    return *m_store.template get_at_index<1>();
+    if (!has_undefined_state())
```

Comment:
Code duplication! Please use here `return std::move(*this).get_error()`. In this case move is used to transform this from an lvalue into a rvalue and nothing is moved behind the scenes!